App.js

```diff
@@ -1,77 +1,48 @@
-import {StyleSheet, View, ScrollView, RefreshControl} from 'react-native';
-import {NavBar} from './src/components/Nav-bar';
-import {NavBarDetails} from './src/components/Nav-bar-details';
-import {SearchInput} from './src/components/Search-input';
-import {Card} from './src/components/Card';
-import mockCards from './src/components/mockCards';
-import {CardLarge} from './src/components/Card-Large';
-
-const BASE_URL = 'https://demo.spreecommerce.org';
-const API_PATH = `${BASE_URL}/api/v2/storefront/products`;
-
-const wait = timeout => {
-  return new Promise(resolve => setTimeout(resolve, timeout));
-};
+import React from 'react';
+import {NavigationContainer} from '@react-navigation/native';
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+import {MainScreen} from './src/pages/Main-screen';
+import {Product} from './src/pages/Product';
+import {createDrawerNavigator} from '@react-navigation/drawer';
+import {ProductAdded} from './src/modal/Product-added';
+import {ChooseColor} from './src/modal/Choose-color';
+import {LoginToContinue} from './src/modal/Login-to-continue';
+import {RegistredCart} from './src/components/Registered-cart';
+import {UnregisteredCart} from './src/components/Unregistered-cart';
+
+const Stack = createNativeStackNavigator();
+const Drawer = createDrawerNavigator();
 
 const App = () => {
-  const [backData, getBackData] = useState();
-
-  useEffect(() => {
-    fetch(API_PATH)
-      .then(response => response.json())
-      .then(data => getBackData(data));
-  }, []);
-
-  const RenderCards = backData?.data.map(item => (
-    <Card
-      key={item.attributes.name}
-      data={item.attributes}
-      icon={{uri: 'https://picsum.photos/100'}}
-    />
-  ));
-
-  const [refreshing, setRefreshing] = React.useState(false);
-
-  const onRefresh = React.useCallback(() => {
-    setRefreshing(true);
-    wait(2000).then(() => setRefreshing(false));
-  }, []);
-
+  // User registration has not yet been implemented
+  let registered = true;
   return (
-    <ScrollView
-      contentContainerStyle={{alignItems: 'center'}}
-      refreshControl={
-        <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
-      }
-      style={styles.main}>
-      <NavBar />
-      <SearchInput />
-      <View style={styles.cardsWrapper}>{RenderCards}</View>
-    </ScrollView>
-    //============================
-    // <ScrollView
-    //   style={styles.main}
-    //   contentContainerStyle={{alignItems: 'center'}}>
-    //   <NavBarDetails />
-    //   <View style={styles.cardsWrapper}>
-    //     <CardLarge data={mockCards[0]} />
-    //   </View>
-    // </ScrollView>
+    <NavigationContainer>
+      <Drawer.Navigator
+        initialRouteName="MainScreen"
+        screenOptions={{
+          headerShown: false,
+        }}>
+        <Drawer.Group>
+          <Drawer.Screen name="MainScreen" component={MainScreen} />
+          <Drawer.Screen name="Product" component={Product} />
```

Comment:
Here the main idea was to use Stack navigator for Main -> Details.  
Using this approach you will have a support of GoBack button from default and also transitions support.  
So my proposal is to use Stack as a global navigator (for main -> details) and Drawer navigator inside. But this is a good example if you want to have drawer only on Main page